### PR TITLE
N function in pedigree sims

### PR DESCRIPTION
**Pull request by hannesbecher**
When simulating pedigrees, one needs to set a population size _N_. This PR adds functionality for _N_ to be a function of the generation (back in time). This works for forward and backward pedigree simulations.

### CODE REVIEW

**Comment by jeromekelleher** [comment]
LGTM - just needs some light test coverage I guess.

Some ideas:
- passing in ``population_size=lamba _: N`` should be identical to ``population_size=N``
- passing in ``population_size=lamba t: 100 t < 10 else 1`` should cause an immediate bottleneck at time 10

**Comment by hannesbecher** [comment]
> LGTM - just needs some light test coverage I guess.
> 
> Some ideas:
> 
>     * passing in `population_size=lamba _: N` should be identical to `population_size=N`
> 
>     * passing in `population_size=lamba t: 100 t < 10 else 1` should cause an immediate bottleneck at time 10

I've tried. Do the added tests make sense?

File: msprime/pedigrees.py [hunk 1]
```diff
@@ -457,6 +458,31 @@ def sim_pedigree(
     end_time=None,
     direction="forward",
 ):
+    """
+    Simulates a pedigree
+
+    Arguments:
```
Comment (jeromekelleher) [inline msprime/pedigrees.py:464]: Can you reformat this using sphinx :param: sections like the rest of the code please?

**Review by jeromekelleher — COMMENTED** [review]
Nice, just need to reformat the docstring style.

**Comment by jeromekelleher** [comment]
Great! Can you squash down to one commit please? (See dev docs for guidance)

**Comment by hannesbecher** [comment]
NB, squashing is mentioned only in the tskit dev docs, I think.

**Pull request merged**